src/portable/synopsys/dwc2/dwc2_stm32.h

```diff
@@ -268,6 +268,77 @@ static inline void dwc2_phy_update(dwc2_regs_t* dwc2, uint8_t hs_phy_type) {
   }
 }
 
+//------------- DCache -------------//
+#if CFG_TUD_MEM_DCACHE_ENABLE || CFG_TUH_MEM_DCACHE_ENABLE
+
+typedef struct
+{
+  uintptr_t start;
+  uintptr_t end;
+} mem_region_t;
+
+#if CFG_TUSB_MCU == OPT_MCU_STM32H7
+  static mem_region_t uncached_regions[] = {
+    // DTCM (although USB DMA can't transfer to/from DTCM)
+    {.start = 0x20000000, .end = 0x2001FFFF},
+  };
+#elif CFG_TUSB_MCU == OPT_MCU_STM32H7RS
+  static mem_region_t uncached_regions[] = {
+    // DTCM (although USB DMA can't transfer to/from DTCM)
+    {.start = 0x20000000, .end = 0x2002FFFF},
+  };
+#elif CFG_TUSB_MCU == OPT_MCU_STM32F7
+  static mem_region_t uncached_regions[] = {
+    // DTCM
+    {.start = 0x20000000, .end = 0x2000FFFF},
+  };
+#else
+  #error "Cache maintenance is not supported yet"
+#endif
+
+TU_ATTR_ALWAYS_INLINE static inline uint32_t round_up_to_cache_line_size(uint32_t size) {
+  if (size & (CFG_TUD_MEM_DCACHE_LINE_SIZE-1)) {
+    size = (size & ~(CFG_TUD_MEM_DCACHE_LINE_SIZE-1)) + CFG_TUD_MEM_DCACHE_LINE_SIZE;
+  }
+  return size;
+}
+
+TU_ATTR_ALWAYS_INLINE static inline bool is_cache_mem(uintptr_t addr) {
+  for (unsigned int i = 0; i < TU_ARRAY_SIZE(uncached_regions); i++) {
+    if (addr >= uncached_regions[i].start && addr <= uncached_regions[i].start)
```

Comment:
This should be:
```suggestion
    if (addr >= uncached_regions[i].start && addr <= uncached_regions[i].end)
```
I think?

Reply:
Oops, thank you !